pkg/cmd/roachtest/roachtestutil/mixedversion/README.md

```diff
@@ -198,6 +191,43 @@ Once all hooks are registered, all that is left to do is to actually run the tes
 
 ### Best Practices
 
+#### Mixed-version Hooks and Compatibility
```

Comment:
Moved under `Best Practices` and attempted to clarify why we need `MinimumSupportedVersion` and how to use it.